src/ui/components/results/directanswercomponent.js

```diff
@@ -237,6 +219,112 @@ export default class DirectAnswerComponent extends Component {
     });
   }
 
+  /**
+   * Determines the card that should be used for the given direct answer.
+   *
+   * @param {Object} directAnswer The direct answer state
+   * @returns {string}
+   */
+  _getCard (directAnswer) {
+    if (this._types) {
+      return this._getCardBasedOnTypes(directAnswer);
+    } else if (this._cardOverrides.length > 0) {
+      return this._getCardBasedOnOverrides({
+        directAnswer: directAnswer,
+        overrides: this._cardOverrides,
+        fallback: this._defaultCard
+      });
+    } else {
+      return this._defaultCard;
+    }
+  }
+
+  /**
+   * Determines the card that should be used based on the types option
+   *
+   * @param {Object} directAnswer The direct answer state
+   * @returns {string}
+   */
+  _getCardBasedOnTypes (directAnswer) {
+    return Object.entries(this._types).reduce((cardType, [directAnswerType, typeOptions]) => {
```

Comment:
could we just directly access the correct directAnswerType in this._types instead of looping?

Reply:
ah, ya the reduce is probably overkill here